src/SFML/Window/DRM/InputImpl.cpp

```diff
@@ -409,22 +406,36 @@ bool eventProcess(sf::Event& event)
                         //
                         if (special)
                         {
-                            event.type         = sf::Event::TextEntered;
-                            event.text.unicode = special;
+                            event = sf::Event::TextEntered{special};
                             return true;
                         }
                     }
                     else if (kb != sf::Keyboard::Unknown)
                     {
                         // key down and key up events
                         //
-                        event.type         = inputEvent.value ? sf::Event::KeyPressed : sf::Event::KeyReleased;
-                        event.key.code     = kb;
-                        event.key.scancode = sf::Keyboard::Scan::Unknown; // TODO: not implemented
-                        event.key.alt      = altDown();
-                        event.key.control  = controlDown();
-                        event.key.shift    = shiftDown();
-                        event.key.system   = systemDown();
+                        if (inputEvent.value)
+                        {
+                            sf::Event::KeyPressed keyPressed;
+                            keyPressed.code     = kb;
+                            keyPressed.scancode = sf::Keyboard::Scan::Unknown; // TODO: not implemented
+                            keyPressed.alt      = altDown();
+                            keyPressed.control  = controlDown();
+                            keyPressed.shift    = shiftDown();
+                            keyPressed.system   = systemDown();
```

Comment:
This can be done into a `Event::KeyChanged` instance to lower code duplication with else branch.